numpy/lib/function_base.py

```diff
@@ -4305,80 +4524,148 @@ def _percentile(a, q, axis=None, out=None,
     if axis is None:
         axis = 0
 
-    Nx = ap.shape[axis]
-    indices = q * (Nx - 1)
+    # move axis to -1 for np.vectorize() operations.  Will move back later.
+    ap = np.swapaxes(ap, axis, -1)
+
+    def _sort_by_index(vector, vec_indices):
+        return vector[vec_indices]
+
+    # this func vectorizes sort along axis
+    arraysort = np.vectorize(_sort_by_index, signature='(i),(i)->(i)')
```

Comment:
Not at all, just pointing out another place that that it could be used if it does get merged